### PR TITLE
Migrate kube-scheduler to fallback for events API built into the library

### PR DESCRIPTION
**Pull request by chelseychen**
**What type of PR is this?**

> /kind cleanup

**What this PR does / why we need it**:

Migrate kube-scheduler to use Event fallback in common lib.

**Special notes for your reviewer**:

This PR is still WIP.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: <https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/383-new-event-api-ga-graduation>
```

### CODE REVIEW

**Comment by ahg-g** [comment]
@chelseychen I see there are changes being made outside the scheduler pkg ( ```staging/src/k8s.io/client-go/tools/events/```), can you please have those changes merged first and then submit a separate PR just for the scheduler changes?

**Comment by ahg-g** [comment]
> @chelseychen I see there are changes being made outside the scheduler pkg (`staging/src/k8s.io/client-go/tools/events/`), can you please have those changes merged first and then submit a separate PR just for the scheduler changes?

oh, I see you are doing that already :) ping me when the other PR merges and this one is ready for review.

File: cmd/kube-scheduler/app/options/options.go [hunk 1]
```diff
@@ -247,13 +248,20 @@ func (o *Options) Config() (*schedulerappconfig.Config, error) {
 		return nil, err
 	}
 
-	coreBroadcaster := record.NewBroadcaster()
+	c.EventClient = events.NewEventClient(eventClient)
+
+	// Set CoreClient and CoreBroadcaster for leader election if they are nil
+	// since leader election hasn't been migrated to use the new Event API.
+	if c.EventClient.CoreBroadcaster == nil {
```
Comment (wojtek-t) [inline cmd/kube-scheduler/app/options/options.go:255]: add a TODO to get rid of that once leader election is migrated.

File: cmd/kube-scheduler/app/server.go [hunk 2]
```diff
@@ -158,12 +158,14 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
...
-	if cc.Broadcaster != nil && cc.EventClient != nil {
-		cc.Broadcaster.StartRecordingToSink(ctx.Done())
-	}
-	if cc.CoreBroadcaster != nil && cc.CoreEventClient != nil {
-		cc.CoreBroadcaster.StartRecordingToSink(&corev1.EventSinkImpl{Interface: cc.CoreEventClient.Events("")})
+	cc.EventClient.StartRecordingToSink(ctx.Done())
+
+	// If scheduler is using new Event API, we should manually start core broadcaster
+	// for leader election since it hasn't been migrated yet.
+	if cc.EventClient.V1beta1Client != nil {
```
Comment (wojtek-t) [inline cmd/kube-scheduler/app/server.go:165]: Let's make it part of the fallback in the library

File: cmd/kube-scheduler/app/options/options.go (hunk 1 above)
Comment (wojtek-t) [reply cmd/kube-scheduler/app/options/options.go:255]: Actually I take it back. I don't like the fact that CoreBroadcaster is a public field.
I will comment on the original PR.

Comment (wojtek-t) [reply cmd/kube-scheduler/app/options/options.go:255]: I have a feeling that this won't even compile - EventClient is an interface that doesn't have CoreBroadcaster field...

File: cmd/kube-scheduler/app/server.go (hunk 2 above)
Comment (wojtek-t) [reply cmd/kube-scheduler/app/server.go:165]: See my comment in the original PR.

File: cmd/kube-scheduler/app/options/options.go (hunk 1 above)
Comment (chelseychen) [reply cmd/kube-scheduler/app/options/options.go]: Updated the codes.

File: cmd/kube-scheduler/app/server.go (hunk 2 above)
Comment (chelseychen) [reply cmd/kube-scheduler/app/server.go]: Updated the codes.

**Comment by wojtek-t** [comment]
/ok-to-test

@chelseychen - the original PR has merged, can you please rebase this one on top?

**Comment by chelseychen** [comment]
@ahg-g - #91798 has been merged and I've rebased this PR, so it's ready for review. PTAL, thanks!

**Comment by chelseychen** [comment]
@wojtek-t 
pull-kubernetes-verify failed since `events.EventBroadcasterAdapter` was marked as `Deprecated`, which caused staticcheck failed.
What should be the right way to fix this? Remove `Deprecated` word and instead use a TODO to indicate that for `EventBroadcasterAdapter`, or add `//lint:ignore` to skip staticcheck for that line?

**Comment by wojtek-t** [comment]
> What should be the right way to fix this? Remove Deprecated word and instead use a TODO to indicate that for EventBroadcasterAdapter, or add //lint:ignore to skip staticcheck for that line?

lint ignore is fine it you can make it. if not changing to TODO is an alternative too

**Pull request merged**